### ISSUE TITLE
isAutoActivateOkay() checks wrong character

### ISSUE THREAD

**Issue by DevCharly** [bobbylight/AutoComplete#77]
isAutoActivateOkay() checks the character after the caret, but shouldn't it check the character **before** the caret?

https://github.com/bobbylight/AutoComplete/blob/da75dd9c0362ec49c64c6d28338ffa0e892873c8/AutoComplete/src/main/java/org/fife/ui/autocomplete/CompletionProviderBase.java#L166-L177

**Comment by bobbylight** [comment]
This does seem wrong, but IIRC this is called in a `DocumentListener#insertUpdate()` callback that occurs before the caret is notified and has its position updated.  Thus `tc.getCaretPosition()` still returns the prior caret position (this code is also only run if the text inserted is 1 character long, so say pasting a text snippet will never trigger the auto-activation, only typing a single character or pasting a single character).

Yeah, this code is a little obtuse :)

**Comment by DevCharly** [comment]
Glad you like FlatLaf 😄  Tried to make it as easy to use in existing apps as possible...

Thanks for the explanation. That makes sense.

I asked because in [FlatLaf Theme Editor](https://github.com/JFormDesigner/FlatLaf/tree/master/flatlaf-theme-editor) the caret position is already updated and `CompletionProviderBase.isAutoActivateOkay()` did not work.

So I've overridden `isAutoActivateOkay()` to check character before caret and now it works fine: [FlatCompletionProvider.java](https://github.com/JFormDesigner/FlatLaf/blob/c404a0d1a922cbaf289ee4b43032208f73b20cb7/flatlaf-theme-editor/src/main/java/com/formdev/flatlaf/themeeditor/FlatCompletionProvider.java#L215-L227)

On the other hand, I've just enabled auto-activation in class `AutoCompleteDemo` (from the README) and there the caret is before the inserted character.

So I tried to figure out what's different and found that my app invokes [TextEditorPane.load(FileLocation loc, String defaultEnc)](https://github.com/bobbylight/RSyntaxTextArea/blob/f03e87801156d406a2e922a831213e36c4314396/RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/TextEditorPane.java#L480) and this changes the behavior.

Maybe loading a file, which creates a new document, changes order of document listeners?

**Comment by bobbylight** [comment]
Yeah, relying on the order of the listeners is pretty bad.  I'll take a
look at fixing things, just have to have a think about the best way to do
it.
